fastlane/Fastfile

```diff
@@ -98,31 +122,22 @@ in xcode user defined build configuration and just reflected in `Info.plist` dur
     )
 
     desc "Read current version number from project configuration."
-    current_version_version = read_project_property(
-      build_configuration: build_configurations[:testflight],
-      build_setting: 'CURRENT_VERSION_NUMBER'
-    )
+    current_version_version = latest_testflight_version(bundle_id: bundle_identifier)
 
-    desc "Read current build number from project configuration."
-    current_build_version = read_project_property(
-      build_configuration: build_configurations[:testflight],
-      build_setting: 'CURRENT_BUILD_NUMBER'
+    desc "Read current build number from TestFlight."
+    current_build_version = latest_testflight_build_number(
+      app_identifier: bundle_identifier,
+      version: current_version_version,
+      initial_build_number: first_build
     )
 
-    desc "Get the type of release desired"
-    bump_type = options[:bump_type]
-    while not release_bump_types.include? bump_type do
-      bump_type = UI.input "Choose the `bump_type` representing the type of deploy. It can be any of %s" % release_bump_types.to_s
-    end
+    bump_type = check_bump_type(
+      build_configuration: build_configuration,
```

Comment:
I'd move this just after checking the bump type for 2 reasons:

- Everything related to that is closer in the code, and makes it easier to be understood.
- I guess reading the version and build time form TF takes a while, let's do it if we are sure it will be used.

Reply:
Can't do that, this needs the current version value so it needs to be after I get it.